azure-pipelines-docs-integration.yml

```diff
@@ -52,13 +52,27 @@ stages:
                 tests/integration/fixtures/**
                 tests/test_sets/**
 
+          - bash: |
+              CHANGED_FILES=$(git diff HEAD origin/develop --name-only)
+              DEPENDENCIES=$(python scripts/trace_docs_deps.py docs)
+              printf '%s\n' "[CHANGED FILES]" "${CHANGED_FILES[@]}" "" "[DEPENDENCIES]" "${DEPENDENCIES[@]}" ""
+
+              echo "Checking for file changes from 'great_expectations/' that impact 'docs/'..."
+              for FILE in $DEPENDENCIES; do
+                if [[ ${CHANGED_FILES[@]} =~ $FILE ]]; then
+                    echo "Found change in local dependency:" $FILE
+                    echo "##vso[task.setvariable variable=DocsDependenciesChanged;isOutput=true]true"
+                    break
+                fi
+              done
+            name: CheckDocsDependenciesChanges
```

Comment:
Azure does not provided native support for arrays so we need to pipe the output of the script to a bash array and iterate through. 

`CHANGED_FILE` compares against `develop` to determine which files have changed. If one of those files is in the output of `trace_docs_deps.py`, the pipeline is triggered.

Should this script be moved to an external bash script? I thought it was small enough to keep in-line but happy to adjust if need be.